plugins/polio/README.md

```diff
@@ -1,28 +1,217 @@
+access the Google Spreadsheet API using a service principal. Service principals
+are a type of identity that represent applications or services rather than
+individual users. By following the steps below, you'll be able to obtain the
+necessary credentials to authenticate your application and access the Google
+Spreadsheet API programmatically.
+
+Step 1: Set up a Google Cloud Platform (GCP) Project:
+
+1. Go to the Google Cloud Console (https://console.cloud.google.com/) and sign
+   in with your Google account.
+2. Create a new project by clicking on the project dropdown menu at the top of
+   the console and selecting "New Project." Provide a name for your project and
+   click "Create."
+3. Once the project is created, ensure that it is selected in the project
+   dropdown menu.
+
+Step 2: Enable the Google Spreadsheet API:
+
+1. In the Google Cloud Console, navigate to the "APIs & Services" -> "Library"
+   section using the left sidebar.
+2. Search for "Google Spreadsheet API" in the library search bar.
+3. Click on the "Google Spreadsheet API" from the search results.
+4. On the API details page, click the "Enable" button to enable the API for your
+   project.
+
+Step 3: Create a Service Account
+
+1. Open the Google Cloud Console (console.cloud.google.com).
+2. Select your desired project from the project selector.
+3. In the left sidebar, navigate to "IAM & Admin" -> "Service accounts."
+4. Click on the "Create Service Account" button.
+5. Provide a name and optional description for the service account.
+6. Click on the "Create" button.
+7. On the "Service account permissions" page, assign the necessary roles for
+   accessing the Google Spreadsheet API. Consider granting the "Editor" role if
+   you require read and write access to spreadsheets.
+8. Click on the "Continue" button.
+9. Skip the "Grant users access to this service account" section, unless
+   specific users need to be granted access.
+10. Click on the "Done" button.
+
+Step 4: Generate a Private Key
+
+1. Locate the newly created service account in the list and click on the "Manage
+   Keys" button in the "Actions" column.
+2. Click on the "Add Key" button and select the "Create new key" option.
+3. Choose the key type as "JSON" and click on the "Create" button.
+4. A JSON file containing the private key and other details will be downloaded
+   to your computer. Make sure to store it securely.
+
+Step 6: Configure it in Iaso
+
+Go to the Django admin at `/admin` and go to Polio -> Config and create a
+olio.Config object with the slug google_api_key and copy the key file (json)
+inside the content field.
+
+# Generation via template
+
+For the generation of Preparedness template you will also need a
+Config `preparedness_template_id` with keys `fr` and `en`.
+
+
+## Generation procedure
+The file is generated for a campaign's round in particular.
+1. Find the template according to the lang of the country where the Campaign occur
+2. Make a copy of it in a new file
```

Comment:
Maybe we can highlight this one as important, to avoid directly modifying the template